### ISSUE TITLE
isotype controls

### ISSUE THREAD

**Issue by igordot** [MattPM/dsb#2]
Currently, it looks like you are combining all the isotype controls as background. Have you considered taking into account which specific antibody each one is matched with? Obviously, that would complicate the process, but is that something that is perhaps on the roadmap?

**Comment by MattPM** [comment]
Hi @igordot , that is an interesting idea, but not something that we pursued here based on how the function is set up and our observations of where the main sources of noise were in CITE-seq data.  The correction applied with `denoise.counts = TRUE`  and `use.isotype.control = TRUE` corrects *each cell* for what we defined in the paper as the "technical component" of that cell's ADT library, which serves to capture technical potentially stochastic per droplet differences in RT efficiency, non-specific binding, etc. One readout of the technical component is the mean expression of the non-expressed proteins for a given cell (the "background mean" of the gaussian mixture model fit on each cell). We found that the background mean correlated with counts in isotype controls across single cells. Based on the correlation of the 4 isotype controls + the background mean we derive the primary latent component (PC1) for each cell and correct the counts using this scalar as a covariate. 

Since we observed the main source of noise was ambient antibody rather than non-specific binding (see Fig 1A), we correct each protein (maybe as you are suggesting doing with the isotypes) by rescaling based on the observed counts in empty droplets.   With flow data, typically the isotype controls give a qualitative measure of the amount of fc binding but it is controversial whether they should be used to guide positivity or gating, for example see: https://www.ncbi.nlm.nih.gov/pubmed/27731950. Also as you alluded to, matching each protein with its isotype control would require a lot of manual set up on the user end, it would be experiment / panel specific and would become more difficult as the size of these panels increases to hundreds of markers, so matching the isotypes to their corresponding markers is likely not something we would add to this function.
